drivers/spi/spi_shell.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2024 Astroligt
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <string.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/sys/util.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_shell, CONFIG_SPI_LOG_LEVEL);
+
+#define ARGV_DEV	(1)
+#define ARGV_BYTES	(2)
+
+/* Maximum bytes we can write and read at once */
+#define MAX_SPI_BYTES	MIN((CONFIG_SHELL_ARGC_MAX - ARGV_BYTES), 32)
+
+static uint32_t frequency = 1000000;
+
+static void device_name_get(size_t idx, struct shell_static_entry *entry)
+{
+	const struct device *dev = shell_device_lookup(idx, "spi");
+
+	entry->syntax = (dev != NULL) ? dev->name : NULL;
+	entry->handler = NULL;
+	entry->help = NULL;
+	entry->subcmd = NULL;
+}
+
+SHELL_DYNAMIC_CMD_CREATE(dsub_device_name, device_name_get);
+
+static int cmd_spi_transceive(const struct shell *shell_ctx,
+			     size_t argc, char **argv)
+{
+	const struct device *dev;
+	uint8_t rx_buffer[MAX_SPI_BYTES] = {0};
+	uint8_t tx_buffer[MAX_SPI_BYTES] = {0};
+
+	dev = device_get_binding(argv[ARGV_DEV]);
+
```

Comment:
you can remove that empty line (unless you remove line 41 and declare dev here, then you have to keep this empty line)